monai/inferers/merger.py

```diff
@@ -0,0 +1,162 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from collections.abc import Sequence
+from typing import Any
+
+import torch
+
+from monai.utils import ensure_tuple_size
+
+__all__ = ["Merger", "AvgMerger"]
+
+
+class Merger(ABC):
+    """
+    A base class for merging patches.
+    Extend this class to support operations for `PatchInference`.
+    There are two methods that must be implemented in the concrete classes:
+
+        - aggregate: aggregate the values at their corresponding locations
+        - finalize: perform any final process and return the merged output
+
+    Args:
+        output_shape: the shape of the merged output tensor. Default to None.
+        device: the device where Merger tensors should reside.
+    """
+
+    def __init__(self, output_shape: Sequence[int] | None = None, device: torch.device | str | None = None) -> None:
+        self.output_shape = output_shape
+        self.device = device
+        self.is_finalized = False
+
+    @abstractmethod
+    def aggregate(self, values: torch.Tensor, location: Sequence[int]) -> None:
```

Comment:
Seems this `aggregate()` API is called multiple times, each time to append 1 patch, and the `finalize()` API actually does the real "aggregation" work.
I would suggest to change it to "append" and "finalize", it's also consistent with other MONAI existing module:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/metrics/cumulative_average.py#L80
What do you think?

Thanks.

Reply:
I see your point but `aggregate` as a verb means ["to collect or gather into a mass or whole"](https://www.merriam-webster.com/dictionary/aggregate), so it correctly define the process of collecting values into a result tensor.

Also consider that this API is common for other [not yet implemented] mergers so these methods needs to be generalizable enough. For instance, for `MaxMerger` that aggregates the maximum value instead of average, all the work is happening in `aggregate` method and `finalize` won't do any thing, so I believe replacing it with append and aggregate won't be appropriate.

Reply:
Hi @ericspod @wyli ,

What do you think about this naming question? I am not very sure about it.

Thanks in advance.

Reply:
I'm fine with the naming as-is. The `aggregate` call in the metrics is collecting data across processes, similarly here `aggregate` means to collect data from sources. The `finalize` part could be a no-op in some cases like @drbeh mentions so `aggregate` then would be doing the real work of combining results like the name would imply.

Reply:
OK, let's keep the current implementation.

Thanks for the discussion.